app/controllers/concerns/atmosphere/api/application_controller_ext.rb

```diff
@@ -25,6 +29,15 @@ def token
           request.headers[Devise::Strategies::TokenAuthenticatable.header_key].
             presence
       end
+
+      def set_project
+        current_user.project = project if current_user
+      end
+
+      def project
+        params[Air.config.project_key] ||
+          request.headers[Air.config.header_project_key]
```

Comment:
Okay; do you know when CLEW and Taverna will actually begin sending these headers?